Task7 - Vertex AI qwiklab

lukyrasocha opened the pull request:
I basically recreated the qwiklab but managed to run everything inside a jupyter notebook

I did this for two reasons:
- I think it is nicer to have everything centralised inside this repo instead of pointing out to a qwiklab
- I find it helpful for my sake to write down how to reproduce what I did and how I solved the errors that I encountered on the way (I hope it's okay)

closes #6

lukyrasocha committed:
Began writing the tutorial

lukyrasocha committed:
Files copied from qwiklab to be reproduced locally in jupyter notebook

lukyrasocha committed:
update

lukyrasocha committed:
update

lukyrasocha committed:
update

lukyrasocha committed:
update

lukyrasocha committed:
update

lukyrasocha committed:
delete

halvgaard commented on Task7/README.md:
````diff
@@ -0,0 +1,135 @@
+gcloud config set project {YOUR_PROJECT_ID}
+```
+
+### 4. Enable cyloud services for your project
````
```suggestion
### 4. Enable cloud services for your project
```

halvgaard commented on Task7/README.md:
````diff
@@ -0,0 +1,135 @@
+    --role="roles/storage.admin"
+```
+
+5.3. Grant it access to your BigQUery data source to read data into your TensorFlow model
````
```suggestion
5.3. Grant it access to your BigQuery data source to read data into your TensorFlow model
```

halvgaard commented on Task7/vertex-ai-qwikstart/online-retail-clv-3M/cloudbuild.yaml:
```diff
@@ -0,0 +1,5 @@
+- name: 'gcr.io/cloud-builders/docker'
+  args: [ 'build', '-t', 'us-central1-docker.pkg.dev/vertex-ai-project2/online-retail-clv/dnn-regressor:latest', '.' ]
+images: 
+- 'us-central1-docker.pkg.dev/vertex-ai-project2/online-retail-clv/dnn-regressor:latest'
```
Are you using US region in everything?

lukyrasocha replied:
That was the default set by the qwiklab, now fixed to europe-west1 and made sure that the notebook still runs as expected

halvgaard commented on Task7/vertex-ai-qwikstart/requirements.txt:
```diff
@@ -0,0 +1,4 @@
+explainable-ai-sdk==1.3.0
+tensorflow-io==0.15.0
+pyarrow
+
```
Rename requirements.txt and requirements2.txt to something meaningful or explain when to use

halvgaard commented on Task7/vertex-ai-qwikstart/utils/data_download.py:
```diff
@@ -0,0 +1,188 @@
+    args:
+    """
+    
+    #set GCS client.
```
```suggestion
```

halvgaard replied:
Please remove other unneeded noisy comments

halvgaard commented on Task7/vertex-ai-qwikstart/utils/data_download.py:
```diff
@@ -0,0 +1,188 @@
+    blob = bucket.blob("data/online_retail.csv")
+
+    #See if file already exists.
+    if blob.exists() == False:
+        try:
+            os.mkdir(LOCAL_PATH)
+            logging.info('Downloading xlsx file...')            
+            local_xlsx = wget.download(args.URL, out=f"{LOCAL_PATH}/{FILENAME}.xlsx")
+            logging.info('Converting xlsx -> csv...')            
+            df = pd.read_excel(local_xlsx)            
+            df.to_csv(f"{LOCAL_PATH}/{FILENAME}.csv", index=False)
+            logging.info('Uploading local csv file to GCS...')
+            blob.upload_from_filename(f"{LOCAL_PATH}/{FILENAME}.csv")
+            logging.info('Copied local csv file to GCS.')
+            # Delete all contents of a directory using shutil.rmtree() and  handle exceptions.
+            try:
+                shutil.rmtree(LOCAL_PATH)
+                logging.info('Cleaning up local tmp data directory...')
+            except:
+                logging.error('Error while deleting local tmp data directory.')
+
+        #print error if file doesn't exist.
+        except BaseException as error:
+            logging.error('An exception occurred: {}'.format(error))
+            
+    #print error if file already exists in GCS.
+    else:
+        logging.warning('File already exists in GCS.')
```
```suggestion
    if blob.exists():
        logging.warning('File already exists in GCS.')
        # TODO fix indent
      try:
          os.mkdir(LOCAL_PATH)
          logging.info('Downloading xlsx file...')            
          local_xlsx = wget.download(args.URL, out=f"{LOCAL_PATH}/{FILENAME}.xlsx")
          logging.info('Converting xlsx -> csv...')            
          df = pd.read_excel(local_xlsx)            
          df.to_csv(f"{LOCAL_PATH}/{FILENAME}.csv", index=False)
          logging.info('Uploading local csv file to GCS...')
          blob.upload_from_filename(f"{LOCAL_PATH}/{FILENAME}.csv")
          logging.info('Copied local csv file to GCS.')
          # Delete all contents of a directory using shutil.rmtree() and  handle exceptions.
          try:
              shutil.rmtree(LOCAL_PATH)
              logging.info('Cleaning up local tmp data directory...')
          except:
              logging.error('Error while deleting local tmp data directory.')

      #print error if file doesn't exist.
      except BaseException as error:
          logging.error('An exception occurred: {}'.format(error))
            

```

lukyrasocha replied:
I am not sure I get this logic
- If the bucket does not exist yet in the Cloud Storage, it will fail
- If the bucket exists, then we will overwrite it with

lukyrasocha replied:
I am not sure I understand this logic, so if the bucket doesn't yet exist, we just skip the whole `if` statement and no new bucket is created

halvgaard replied:
I'm missing an indent I see. I was just trying to move the short else block with the warning up to avoid these large if-else blocks. Should the try block be run always or only if the file does not exist?

lukyrasocha replied:
The try block should only be run if the file does not exist. (If the file does not yet exist in GCS, we download it from a given URL and then upload it)

halvgaard commented on Task7/vertex-ai-qwikstart/utils/data_download.py:
```diff
@@ -0,0 +1,188 @@
+    args:
+    schema:
+    """    
+    # Construct a BigQuery client object.
```
```suggestion
```

halvgaard commented on Task7/vertex-ai-qwikstart/utils/data_download.py:
```diff
@@ -0,0 +1,188 @@
+    
+
+def make_dataset_clean_bq(args, query: str):
+    """
+    args:
+    query:
+    """
```
Use doc strings or delete them

halvgaard commented on Task7/vertex-ai-qwikstart/utils/data_download.py:
```diff
@@ -0,0 +1,188 @@
+    logging.info('BQ make clean dataset starting...')
+    try:
+      job = client.query(clean_query)
+      _ = job.result()
```
Why return?

halvgaard replied:
Minor thin: Why not just job.result()? No need to return to a single _ ?

lukyrasocha replied:
Oh, I see. I will change it

halvgaard commented on Task7/vertex-ai-qwikstart/utils/dataset_clean.py:
```diff
@@ -0,0 +1,49 @@
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""TODO."""
```
```suggestion
```

halvgaard commented on Task7/vertex-ai-qwikstart/utils/dataset_clean.py:
```diff
@@ -0,0 +1,49 @@
+# limitations under the License.
+"""TODO."""
+
+dataset_clean_query = """
```
Constant? Upper case

halvgaard reviewed (changes requested):
Good job.
I added some baisc code comments before running the content.
Also ignore binaries?
What region are you using? Docker file seem to suggest US?

lukyrasocha committed:
ignore binaries

lukyrasocha committed:
remove binaries

lukyrasocha committed:
fixed typos

lukyrasocha committed:
update tutorial

lukyrasocha committed:
changed region and made sure that everything runs as expected - actually found a bug in their util function that I needed to fixed - also planning to make a pull request to the original repo

lukyrasocha committed:
fixed small bug in data_download.py, deleted unecessary comments, made some stuff uppercase

lukyrasocha committed:
changed name of requirements that need to be installed to run the notebook correctly

lukyrasocha committed:
small update

lukyrasocha committed:
made table_schema uppercase

lukyrasocha commented:
I resolved all the things you mentioned, here is the summary:
- Ignored binaries
- Fixed typos in the tutorial
- Removed one `requirements.txt` - now we have one requirements for the docker, and one to run the notebook
- Changed region to `europe-west1`
    - here I actually got stuck, because there was a small bug in the utils (the utils were provided by the qwiklab and were cloned from the official repo)
    - But is fixed now, I am planning to make a pull request to the official repo with the fix
- I deleted some of the unnecessary comments - but its quite hard for me to evaluate what should be deleted
- Made constants uppercase

The only thing I didn't resolve is
> Why return?

Because I am not sure how

Please let me know if there are any additional things you'd like me to fix

lukyrasocha committed:
changed the return based on the comment

lukyrasocha commented:
Now all the changes based on the previous comments should be applied 🙂
